src/System.Console/tests/SyncTextReader.cs

```diff
@@ -0,0 +1,206 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.IO;
+using Xunit;
+
+public class SyncTextReaderTests
+{
+    // NOTE: These tests test the underlying SyncTextReader by
+    // accessing the Console.In TextReader (which in fact is a SyncTextReader).
+
+    static readonly string[] s_testLines = new string[] {
```

Comment:
I've not looked at the ReadAndReadLines test.  If there's an explanation there for how this set was arrived at, and if it applies here, seem reasonable to reuse.  I'm just not understanding what the goal is of each variation.  There's lots of repetition, and it seems like all of the call sites just concatenate this all into a single string before using it.  I'd have expected we'd want to test things like:

- dealing with short and long strings, including those that test boundary conditions (e.g. zero, something that would force going being an internal buffer, etc.)
- dealing with a range of characters, including those from a variety of character sets we expect to be supported by the reader
- etc.

It's not clear to me that this does that, or if it's trying to do something else, what that is :smile: I may just be missing it.